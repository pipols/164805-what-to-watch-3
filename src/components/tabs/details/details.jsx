import React from "react";
import PropTypes from "prop-types";

const structure = (data) => [[
  {
    name: `Director`,
    value: data.producer
  },
  {
    name: `Starring`,
    value: data.actors
  }
],
[
  {
    name: `Run time`,
    value: data.duration
  },
  {
    name: `Genre`,
    value: data.genre
  },
  {
    name: `Released`,
    value: data.year
  }
]];

const writeActors = (actors) => actors.map((name) => <> key={name} {name} <br/> </>);

const Details = ({producer, actors, duration, genre, year}) => {
  const data = {producer, actors, duration, genre, year};
  return (
    <div className="movie-card__text movie-card__row">
      {structure(data).map((col, i) => (
        <div key={`col` + i} className="movie-card__text-col">
          {col.map((item) => (
            <p key={item.name} className="movie-card__details-item">
              <strong className="movie-card__details-name">{item.name}</strong>
              <span className="movie-card__details-value">
                {item.name === `Starring` ? writeActors(item.value) : item.value}
              </span>
            </p>
          ))}
        </div>
      ))}
    </div>
  );
};
// //////////////////////////////////////////////////
// const Details = ({producer, actors, duration, genre, year}) => (
//   <div className="movie-card__text movie-card__row">
//     <div className="movie-card__text-col">
//       <p className="movie-card__details-item">
//         <strong className="movie-card__details-name">Director</strong>
//         <span className="movie-card__details-value">{producer}</span>
//       </p>
//       <p className="movie-card__details-item">
//         <strong className="movie-card__details-name">Starring</strong>
//         <span className="movie-card__details-value">
//           {actors.map((name) => <> {name} <br/> </>)}
//         </span>
//       </p>
//     </div>
//
//     <div className="movie-card__text-col">
//       <p className="movie-card__details-item">
//         <strong className="movie-card__details-name">Run Time</strong>
//         <span className="movie-card__details-value">{duration}</span>
//       </p>
//       <p className="movie-card__details-item">
//         <strong className="movie-card__details-name">Genre</strong>
//         <span className="movie-card__details-value">{genre}</span>
//       </p>
//       <p className="movie-card__details-item">
//         <strong className="movie-card__details-name">Released</strong>
//         <span className="movie-card__details-value">{year}</span>
//       </p>
//     </div>
//   </div>
// );

Details.propTypes = {
  producer: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  duration: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired
};

export default Details;
