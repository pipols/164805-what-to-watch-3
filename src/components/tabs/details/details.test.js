import React from "react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import NameSpace from "../../../reducer/name-space";
import {Router} from "react-router-dom";
import history from "../../../history";
import {Details} from "./details.jsx";

const mockStore = configureStore([]);

const film = {
  title: `The Grand Budapest Hotel`,
  cover: `the-grand-budapest-hotel-poster.jpg`,
  previewImage: `the-grand-budapest-hotel-poster.jpg`,
  poster: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#fff`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H.`,
  rating: 8.9,
  votes: 240,
  producer: `Wes Andreson`,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ],
  duration: 120,
  genre: `Drama`,
  year: 2014,
  id: 1,
  isFavorite: true,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
};

const store = mockStore({
  [NameSpace.DATA]: {
    films: [film],
  },
  [NameSpace.STATE]: {
    id: 1
  }
});

it(`<Details /> renders correctly`, () => {
  const tree = renderer
  .create(
      <Provider store={store}>
        <Router history={history} >
          <Details film={film} />
        </Router>
      </Provider>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
