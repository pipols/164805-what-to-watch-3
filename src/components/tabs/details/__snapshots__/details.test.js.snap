// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Wes Andreson
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
         key=
        Bill Murray
         
        Bill Murray
         
        <br />
         
         key=
        Edward Norton
         
        Edward Norton
         
        <br />
         
         key=
        Jude Law
         
        Jude Law
         
        <br />
         
         key=
        Willem Dafoe
         
        Willem Dafoe
         
        <br />
         
         key=
        Saoirse Ronan
         
        Saoirse Ronan
         
        <br />
         
         key=
        Tony Revoloru
         
        Tony Revoloru
         
        <br />
         
         key=
        Tilda Swinton
         
        Tilda Swinton
         
        <br />
         
         key=
        Tom Wilkinson
         
        Tom Wilkinson
         
        <br />
         
         key=
        Owen Wilkinson
         
        Owen Wilkinson
         
        <br />
         
         key=
        Adrien Brody
         
        Adrien Brody
         
        <br />
         
         key=
        Ralph Fiennes
         
        Ralph Fiennes
         
        <br />
         
         key=
        Jeff Goldblum
         
        Jeff Goldblum
         
        <br />
         
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 39m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Drama
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2014
      </span>
    </p>
  </div>
</div>
`;
